[DeadStoreElimination] Introduce test for #63473 (NFC)

antoniofrighetto opened the pull request:
Preliminary tests addition.

antoniofrighetto committed:
[DeadStoreElimination] Introduce test for #63473 (NFC)

antoniofrighetto committed:
!fixup add test4

fhahn commented on llvm/test/Transforms/DeadStoreElimination/dependencies-cross-iterations.ll:
```diff
@@ -0,0 +1,164 @@
+
+loop:                                                ; preds = %loop, %entry
+  %iv = phi i64 [ 0, %entry ], [ %iv.next, %loop ]
+  %ptr.iv = getelementptr inbounds [20 x i64], ptr %array, i64 0, i64 %iv
```
Can define this outside?

fhahn reviewed (commented):
Would be good to also add a test case with cross-iteration dependencies in the loops?

antoniofrighetto committed:
!fixup move tests in read-clobber-after-overwrite.ll

antoniofrighetto committed:
!fixup add cross-iteration dependencies tests

antoniofrighetto commented:
> Would be good to also add a test case with cross-iteration dependencies in the loops?

Reorganized the tests a bit, moved the read after clobbering out in `read-clobber-after-overwrite.ll`, added a few actual cross-iteration dependencies tests; thanks!
